modules/services/home-manager-auto-upgrade.nix

```diff
@@ -61,7 +89,7 @@ in {
             echo "Update Nix's channels"
             ${pkgs.nix}/bin/nix-channel --update
             echo "Upgrade Home Manager"
-            ${homeManagerPackage}/bin/home-manager switch
+            ${homeManagerPackage}/bin/home-manager switch ${toString cfg.flags}
```

Comment:
```suggestion
            ${homeManagerPackage}/bin/home-manager switch ${lib.escapeShellArgs cfg.flags}
```

Reply:
It is intentional as `lib.escapeShellArgs` would produce
```
/nix/store/xxx-home-manager/bin/home-manager switch '--flake github:alice/dotfiles'
```
Which doesn't work, the intent is to produce:
```
/nix/store/xxx-home-manager/bin/home-manager switch --flake github:alice/dotfiles
```

Reply:
So you should use `[ "--flake" (cfg.flake) ]`.

Reply:
On its own, it won't make much difference, but if we combine the two, we'll get the desired results 😊
```
[ "--flake" (lib.escapeShellArg cfg.flake) ] 
```